docs/marketplace/dev/technical-enhancement/migration-guide-upgrade-to-angular-v15.md

```diff
@@ -0,0 +1,442 @@
+
+This document provides instructions for upgrading Angular to version 15 in your Spryker project.
+
+## Overview
+
+Our current version of Angular is v12 with v9 compatibility.
+Angular v12 was deprecated on 2022-11-12 according to the [policy](https://angular.io/guide/releases#support-policy-and-schedule).
+The current stable version of Angular is 15.
+We need to upgrade to the latest major to get the latest bug fixes and security fixes as well as performance in runtime and tooling.
+We can't maintain compatibility with older versions of Angular when updating Angular v15.
+So the update will require a major release for these modules:
+
+- ZedUi
+- DashboardMerchantPortalGui
+- GuiTable
+- MerchantProfileMerchantPortalGui
+- ProductMerchantPortalGui
+- ProductOfferMerchantPortalGui
+- SalesMerchantPortalGui
+- SecurityMerchantPortalGui
+- UserMerchantPortalGui
+
+*Estimated migration time: 2h*
+
+### 1) Update modules
+
+1. Upgrade modules to the new version:
+
+The marketplace modules must correspond to the following versions:
+
+| NAME                                        | VERSION   |
+| ------------------------------------------- | --------- |
+| DashboardMerchantPortalGui                  | >= 2.0.0  |
+| GuiTable                                    | >= 2.0.0  |
+| MerchantProfileMerchantPortalGui            | >= 2.0.0  |
+| ProductMerchantPortalGui                    | >= 3.0.0  |
+| ProductOfferMerchantPortalGui               | >= 2.0.0  |
+| SalesMerchantPortalGui                      | >= 2.0.0  |
+| SecurityMerchantPortalGui                   | >= 2.0.0  |
+| UserMerchantPortalGui                       | >= 2.0.0  |
+| ZedUi                                       | >= 2.0.0  |
+
+If not, update module versions manually or by using the following command:
+
+```bash
+composer update spryker/dashboard-merchant-portal-gui spryker/gui-table spryker/merchant-profile-merchant-portal-gui spryker/product-merchant-portal-gui spryker/product-offer-merchant-portal-gui spryker/sales-merchant-portal-gui spryker/security-merchant-portal-gui spryker/user-merchant-portal-gui spryker/zed-ui
+```
+
+2. Regenerate the data transfer object:
+
+```bash
+console transfer:generate
+```
+
+## 2) Update Webpack
+
+Before starting migration it's required to update Webpack version with related dependencies and configurations.
+Here is the link to [Webpack migration guide](/docs/scos/dev/front-end-development/migration-guide-upgrade-to-webpack-v5.html).
+
+## 3) Update NPM dependencies
+
+{% info_block infoBox "Info" %}
+
+Make sure you are using [Node 16 or later](https://nodejs.org/dist/latest-v16.x/).
+
+{% endinfo_block %}
+
+1. In the `package.json` update or add the following dependencies:
+
+```json
+{
+    "dependencies": {
+        "@angular/animations": "~15.0.3",
+        "@angular/cdk": "~15.0.3",
+        "@angular/elements": "~15.0.3",
+        "@angular/forms": "~15.0.3",
+        "@angular/router": "~15.0.3",
```

Comment:
Please check if the `@angular/router` dependency necessary for us.

Reply:
```
Error: Module not found: Error: Can't resolve '@angular/router' in '/Users/danil.moroz/Projects/nonsplit-docker/project/node_modules/ng-zorro-antd/fesm2020'
```